lib/ci-helper.ts

```diff
@@ -687,6 +688,25 @@ export class CIHelper {
             result = false;
         }
 
+        // if no initial failure, run linting checks
+
+        if (result) {
+            const results = await Promise.all(
+                commits.map((commit: IPRCommit) => {
+                const linter = new LintCommit(commit);
+                return linter.lint();
+            }));
+
+            for (const lintError of results) {
```

Comment:
> I have a hunch that this would look even slicker if you used `.filter()` and `.map()`. For example, I could imagine that filtering out the `void`s could be done via `results.filter((el) => { el })` or some such.

Added `.filter()`.  Not sure creating a new array is more efficient but the code is cleaner.  The problem with `.map()` is the function is not async (and I have not seen a way to do it), so I left the for loop in place.  btw, once you approve this change I would like to add another commit before merging.  It would add the missing async to `addComment()` calls I missed in a previous PR. 😳

> And while we already have all the comments in one place, maybe we want to accumulate them into a single PR comment? Dunno... what do you think?

My concern with making a single comment is that it is easy to ignore parts of the comment.  Putting each commit in a separate comment allows them to be addressed individually.  OTOH, if there is a preference for different treatment based on `push` versus `submit comment`, that could be handled by `addComment()`.  If people complain about too many comments they need to improve their commit messages.  😉

Reply:
> The problem with `.map()` is the function is not async (and I have not seen a way to do it),

I _think_ that `await Promise.all(something.map(...))` would work, but I am pretty new to this `async`/`await` business, I only started using it with GitGitGadget.

>  It would add the missing async to `addComment()` calls I missed in a previous PR.

I just added a commit to that end.

> My concern with making a single comment is that it is easy to ignore parts of the comment.

True.

Reply:
> > The problem with `.map()` is the function is not async (and I have not seen a way to do it),
> 
> I _think_ that `await Promise.all(something.map(...))` would work, but I am pretty new to this `async`/`await` business, I only started using it with GitGitGadget.

I see you added `Promise.all()` to your commit.  I have a concern with it.

1. The function is not returning the promise.  ie it should `return (or await) addComment()` so `Promise.all()` has a promise to wait on.
2. There appears to be a possible object reuse problem (see code below):
2.1 in the map() function addComment() is not run yet because it returns a promise
2.2 this can result in multiple pending addComment()
2.3 Promise.all() causes the addComment() functions to be started
2.4 if an addComment() is suspended in github communications, another addComment() will be started (normal node processing)
2.5 all the addComment() functions share a single GitHubGlue and octokit.  Could this lead to internal state errors in oktokit (two calls at the same time)?

Code to demo 2 above.  In this test, if addComment() is not returned/await'ed, jest gets errors.
```ts
test("Test async map", async () => {
    const addComment = async (body: string) => {
        console.log(`Adding comment to ${body}`);
        await setTimeoutPromise(4000, "foobar"); // pretend GutHubGlue.addPRComment()
        console.log(`Added comment to ${body}`);
    };

    const commits = [ "1", "2", "3"];
    console.log("starting");
    await Promise.all(commits.map(async (cm: string) => {
        return addComment(cm);
    }));
    console.log("promises done");
});
```
Does this make sense?
> I just added a commit to that end.

Thanks!

Reply:
True. So we have to convert to a `for (... of ...) { ... await addComment(...) ...}` loop?

Reply:
Probably the simplest.  The initial code had no async calls.  Are you doing it or shall I?